src-plugins/itkFilters/itkFiltersComponentSizeThresholdProcess_p.h

```diff
@@ -35,86 +35,102 @@ class itkFiltersComponentSizeThresholdProcessPrivate : public itkFiltersProcessB
-        
-            connectedComponentFilter->SetInput ( dynamic_cast<ImageType *> ( ( itk::Object* ) ( input->data() ) ) );
             connectedComponentFilter->Update();
+        }
+        catch( itk::ExceptionObject & err )
+        {
+            qDebug() << "ExceptionObject caught in sizeThresholdingProcess, ConnectedComponentImageFilter!" << err.GetDescription();
+            return DTK_FAILURE;
+        }
 
-            // RELABEL COMPONENTS according to their sizes (0:largest(background))
-            typedef itk::RelabelComponentImageFilter<ImageType, ImageType> FilterType;
-            typename FilterType::Pointer relabelFilter = FilterType::New();
-            relabelFilter->SetInput(connectedComponentFilter->GetOutput());
-            relabelFilter->SetMinimumObjectSize(minimumSize);
+        // RELABEL COMPONENTS according to their sizes (0:largest(background))
+        typedef itk::RelabelComponentImageFilter<ImageType, ImageType> FilterType;
+        typename FilterType::Pointer relabelFilter = FilterType::New();
+        relabelFilter->SetInput(connectedComponentFilter->GetOutput());
+        relabelFilter->SetMinimumObjectSize(minimumSize);
+        try
+        {
             relabelFilter->Update();
+        }
+        catch( itk::ExceptionObject & err )
+        {
+            qDebug() << "ExceptionObject caught in sizeThresholdingProcess, RelabelComponentImageFilter!" << err.GetDescription();
+            return DTK_FAILURE;
+        }
+
+        // BINARY FILTER
+        typedef itk::BinaryThresholdImageFilter <ImageType, ImageType>
+                BinaryThresholdImageFilterType;
 
-            // BINARY FILTER
-            typedef itk::BinaryThresholdImageFilter <ImageType, ImageType>
-            BinaryThresholdImageFilterType;
- 
-            typename BinaryThresholdImageFilterType::Pointer thresholdFilter
-            = BinaryThresholdImageFilterType::New();
-            thresholdFilter->SetInput(relabelFilter->GetOutput());
-            thresholdFilter->SetUpperThreshold(0);
-            thresholdFilter->SetInsideValue(0);
-            thresholdFilter->SetOutsideValue(1);
+        typename BinaryThresholdImageFilterType::Pointer thresholdFilter
+                = BinaryThresholdImageFilterType::New();
+        thresholdFilter->SetInput(relabelFilter->GetOutput());
+        thresholdFilter->SetUpperThreshold(0);
+        thresholdFilter->SetInsideValue(0);
+        thresholdFilter->SetOutsideValue(1);
+        try
+        {
             thresholdFilter->Update();
 
-        
             callback = itk::CStyleCommand::New();
             callback->SetClientData ( ( void * ) this );
             callback->SetCallback ( itkFiltersProcessBasePrivate::eventCallback );
-        
+
             connectedComponentFilter->AddObserver ( itk::ProgressEvent(), callback );
-        
+
             output->setData ( thresholdFilter->GetOutput() );
         }
```

Comment:
hmmm, i hesitated on this one. I don't know if setClientData or setCallBack can crash but they are 2 itk functions, so, we can't be sure.